src/components/NameForm.tsx

```diff
@@ -0,0 +1,40 @@
+import React, {ChangeEventHandler, SyntheticEvent} from "react";
+import ReactDOM from "react-dom";
+
+type Props = { callback: (arg: number) => void };
+type State = { value: number };
+
+class NameForm extends React.Component<Props, State> {
+    handleChange: ChangeEventHandler<HTMLInputElement>;
+    //handleChange: {event: SyntheticEvent<Element, Event>};
+    handleSubmit: () => void;
+    state: { value: number };
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {value: 0};
+
+        this.handleChange = (event) => this.setState({value: parseInt(event.target.value)});
+        this.handleSubmit = () => {
+            props.callback(this.state.value)
```

Comment:
You don't need to use '{}' if you have only one expression.
e.g () => props.callback(this.state.value)